### PR TITLE
ember-data: fix store.query resolve value

### PR DESCRIPTION
**Pull request by BarryThePenguin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://api.emberjs.com/ember-data/release/classes/AdapterPopulatedRecordArray
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

I've always found the return type for `store.query()` confusing. With the changes in #61873 it looks like I was relying on `any` due to `ArrayProxy` extending `EmberObject.extend()`

The docs in [`Store.query`](https://api.emberjs.com/ember-data/3.28/classes/Store/methods/query?anchor=query) state that..

> This method returns a promise, which is resolved with an [AdapterPopulatedRecordArray](https://api.emberjs.com/ember-data/release/classes/AdapterPopulatedRecordArray) once the server returns.

..and from the source I can see that `Store.query` does return a `PromiseArray`..

https://github.com/emberjs/data/blob/fbbff3dabdfcc4ced3d2ecfe8ffe8a4714fbaecb/packages/store/addon/-private/system/core-store.ts#L2037

..which does resolve with a `RecordArray`..

https://github.com/emberjs/data/blob/fbbff3dabdfcc4ced3d2ecfe8ffe8a4714fbaecb/packages/store/addon/-private/system/store/finders.js#L405-L441

I can also see this is the case while running the code in a browser console

### CODE REVIEW

File: types/ember-data/test/store.ts [hunk 1]
```diff
@@ -155,15 +155,22 @@ const tom = store
     });
 
 // GET /users?isAdmin=true
-const admins = store.query('user', { isAdmin: true });
-assertType<DS.AdapterPopulatedRecordArray<User> & DS.PromiseArray<User, Ember.ArrayProxy<User>>>(admins);
+const adminsQuery = store.query('user', { isAdmin: true });
+assertType<DS.PromiseArray<User, DS.AdapterPopulatedRecordArray<User>>>(adminsQuery);
+
+adminsQuery.then(function (admins) {
+    console.log(admins.get("length")); // 42
```
Comment (BarryThePenguin) [inline types/ember-data/test/store.ts:162]: It looks like the original tests came from the example at [`AdapterPopulatedRecordArray`](https://api.emberjs.com/ember-data/release/classes/AdapterPopulatedRecordArray), but I don't think the usage of..

```js
admins.then(function() {
  console.log(admins.get("length")); // 42
});
```

..is correct, as `admins` is already the resolved value from the query

**Comment by sandersn** [comment]
ember types are complex enough that I would like a signoff from a maintainer:  @dwickern, @mike-north, @chriskrycho, @jamescdavis, @dfreeman.

**Review by chriskrycho — APPROVED** [review]
😂 😭 fair, @sandersn. Too real, but fair. Sorry this got missed, @BarryThePenguin! Reviewing

Note for all of us: depending on which order the bot ends up publishing this in, we might have to do a follow-on to make sure the v4 types end up as matching `latest` on npm.

**Comment by chriskrycho** [comment]
Ready to merge

**Comment by BarryThePenguin** [comment]
Thanks everybody 👋🏻

**Pull request merged**